connectors/es/sink.py

```diff
@@ -546,60 +573,53 @@ async def get_access_control_docs(self, generator):
             )
 
         count = 0
-        try:
-            async for doc in generator:
-                doc, _, _ = doc
-                count += 1
-                if count % self.display_every == 0:
-                    self._log_progress()
-
-                doc_id = doc["id"] = doc.pop("_id")
-                doc_exists = doc_id in existing_ids
-
-                if doc_exists:
-                    last_update_timestamp = existing_ids.pop(doc_id)
-                    doc_not_updated = (
-                        TIMESTAMP_FIELD in doc
-                        and last_update_timestamp == doc[TIMESTAMP_FIELD]
-                    )
-
-                    if doc_not_updated:
-                        continue
+        async for doc in generator:
+            doc, _, _ = doc
+            count += 1
+            if count % self.display_every == 0:
+                self._log_progress()
+
+            doc_id = doc["id"] = doc.pop("_id")
+            doc_exists = doc_id in existing_ids
+
+            if doc_exists:
+                last_update_timestamp = existing_ids.pop(doc_id)
+                doc_not_updated = (
+                    TIMESTAMP_FIELD in doc
+                    and last_update_timestamp == doc[TIMESTAMP_FIELD]
+                )
 
-                    self.total_docs_updated += 1
+                if doc_not_updated:
+                    continue
 
-                    operation = OP_UPSERT
-                else:
-                    self.total_docs_created += 1
+                self.total_docs_updated += 1
 
-                    if TIMESTAMP_FIELD not in doc:
-                        doc[TIMESTAMP_FIELD] = iso_utc()
+                operation = OP_UPSERT
+            else:
+                self.total_docs_created += 1
 
-                    operation = OP_INDEX
+                if TIMESTAMP_FIELD not in doc:
+                    doc[TIMESTAMP_FIELD] = iso_utc()
 
-                await self.queue.put(
-                    {
-                        "_op_type": operation,
-                        "_index": self.index,
-                        "_id": doc_id,
-                        "doc": doc,
-                    }
-                )
+                operation = OP_INDEX
 
-                await asyncio.sleep(0)
-        except Exception as e:
-            self._logger.critical("The document fetcher failed", exc_info=True)
-            await self.queue.put("FETCH_ERROR")
-            self.fetch_error = e
-            return
+            await self.put_doc(
+                {
+                    "_op_type": operation,
+                    "_index": self.index,
+                    "_id": doc_id,
+                    "doc": doc,
+                }
+            )
+            await asyncio.sleep(0)
```

Comment:
Do we need `await asyncio.sleep(0)` here? We're already inside an async generator, which should yield control back to the event loop, if possible/needed, right?

Reply:
I'm not adding this line. This is from line 589 (see left).